qutip/core/operators.py

```diff
@@ -639,25 +639,29 @@ def _implicit_tensor_dimensions(dimensions):
     dimensions : list
         Dimension list in the form required by ``Qobj`` creation.
     """
-    if not isinstance(dimensions, list):
+    if not isinstance(dimensions, (list, Space)):
```

Comment:
If this can now accept a Space, should it not also accept a Dimensions object?

Reply:
Dimensions object would change the input meaning causing more confusion:
`qeye([[2], [2]]) != qeye(Dimension([[2], [2]]))`
The first case create a super operator, while the second an operator...

For `qzero` it would be nice since rectangular operators are fine, but changing the input to no longer match `qeye` is probably not a good idea.